product-scenarios/3-securing-an-api-with-different-security-protocols/3.1-api-developers-want-to-secure-their-api-using-an-authentication-machanism/3.1.1-api-developers-and-publishers-want-to-make-their-apis-secure-using-the-oauth2-protocol/src/test/java/org/wso2/am/scenario/tests/api/secure/oauth2/SecureUsingAuth2TestCases.java

```diff
@@ -20,58 +20,142 @@
     private static final String ADMIN_LOGIN_USERNAME = "admin";
     private static final String ADMIN_PASSWORD = "admin";
-    private final String API_DEVELOPER_USERNAME = "3.1.1-user";
-    private final String API_DEVELOPER_PASSWORD = "password@3.1.1-user";
+    private static final String API_DEVELOPER_USERNAME = "3.1.1-developer";
+    private static final String API_DEVELOPER_PASSWORD = "password@123";
     private String backendEndPoint = "http://ws.cdyne.com/phoneverify/phoneverify.asmx";
+    String accessToken;
+    private static final String TEST_API_1_NAME = "PhoneVerifyAPI-1";
+    private static final String TEST_API_2_NAME = "PhoneVerifyAPI-2";
+    private static final String TEST_API_1_CONTEXT = "/phone";
+    private static final String TEST_API_2_CONTEXT = "/phones";
+    private static final String TEST_API_1_VERSION = "1.0.0";
+    private static final String TEST_API_2_VERSION = "1.0.0";
+    private static final String TEST_APPLICATION_NAME = "TestApp1";
 
     @BeforeClass(alwaysRun = true)
     public void init() throws Exception {
         apiStore = new APIStoreRestClient(storeURL);
         apiPublisher = new APIPublisherRestClient(publisherURL);
 
         createUserWithSubscriberRole(SUBSCRIBER_USERNAME, SUBSCRIBER_PASSWORD, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
-        createUserWithPublisherAndCreatorRole(API_DEVELOPER_USERNAME, API_DEVELOPER_PASSWORD, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        createUserWithPublisherAndCreatorRole(API_DEVELOPER_USERNAME, API_DEVELOPER_PASSWORD, ADMIN_LOGIN_USERNAME,
+                ADMIN_PASSWORD);
 
         apiStore.login(SUBSCRIBER_USERNAME, SUBSCRIBER_PASSWORD);
         apiPublisher.login(API_DEVELOPER_USERNAME, API_DEVELOPER_PASSWORD);
+
+        // create and publish sample API
+        String swaggerFilePath = System.getProperty("framework.resource.location") + "swaggerFiles" + File.separator +
+                "phoneverify-swagger.json";
+        File swaggerFile = new File(swaggerFilePath);
+        String swaggerContent = readFromFile(swaggerFile.getAbsolutePath());
+        JSONObject swaggerJson = new JSONObject(swaggerContent);
+
+        apiPublisher.developSampleAPI(swaggerJson, API_DEVELOPER_USERNAME, backendEndPoint, true, "public");
+        createApplication(TEST_APPLICATION_NAME);
+
+        // Check the visibility of the API in API store
+        isAPIVisibleInStore(TEST_API_1_NAME, apiStore);
+
+        // Add subscription to API
+        SubscriptionRequest subscriptionRequest = new SubscriptionRequest(TEST_API_1_NAME, TEST_API_1_VERSION, API_DEVELOPER_USERNAME,
+                TEST_APPLICATION_NAME, APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED);
+        HttpResponse addSubscriptionResponse = apiStore.subscribe(subscriptionRequest);
+        verifyResponse(addSubscriptionResponse);
+        log.info(TEST_APPLICATION_NAME + " is subscribed to " + TEST_API_1_NAME);
+
+        accessToken = generateAppKeys();
```

Comment:
generateAppKeys() method basically returns access token under the client credentials grant type. Doesn't it need to be transferred into the test method (testOAuth2Authorization) since there are other test cases which are needed to obtain valid token under the client credentials as well as password grant types?

Reply:
Yes, it has to be done. This has to be separately implemented as current implementation doesn't have a way to pass the grant type. We may add it in a separate PR as discussed earlier, so that current test implementations won't be blocked on the changes done with this PR